x/wasm/keeper/query_plugins.go

```diff
@@ -269,8 +272,32 @@ func IBCQuerier(wasm contractMetaDataSource, channelKeeper types.ChannelKeeper)
 }
 
 func StargateQuerier(queryRouter GRPCQueryRouter) func(ctx sdk.Context, request *wasmvmtypes.StargateQuery) ([]byte, error) {
-	return func(ctx sdk.Context, msg *wasmvmtypes.StargateQuery) ([]byte, error) {
-		return nil, wasmvmtypes.UnsupportedRequest{Kind: "Stargate queries are disabled."}
+	return func(ctx sdk.Context, request *wasmvmtypes.StargateQuery) ([]byte, error) {
+		binding, whitelisted := StargateLayerBindings.Load(request.Path)
```

Comment:
Can we call this binding? I feel like the semantics are different

In my eyes, bindings are API abstractions that we implement on top of the existing message server and querier.

On the other hand, this seems like a query response to me.

Please correct me if I'm wrong

```suggestion
		response, whitelisted := StargateLayerBindings.Load(request.Path)
```

Reply:
I named it binding because imo it is the layer of API abstraction, as it returns an interface to marshal into, not the response itself